rcl/include/rcl/time.h

```diff
@@ -161,6 +169,15 @@ rcl_clock_valid(rcl_clock_t * clock);
 /**
  * This will allocate all necessary internal structures, and initialize variables.
  *
+ * <hr>
+ * Attribute          | Adherence
+ * ------------------ | -------------
+ * Allocates Memory   | Yes [1]
+ * Thread-Safe        | No
```

Comment:
Can you clarify why the function isn't (same for the others)?

Reply:
I was unsure about this actually, as I have not found an explicit statement of what we mean by thread-safe here. I meant it in the strictest sense i.e. concurrent access to the same object (which is an argument purely because of a language limitation) is safe. If by thread-safe we mean concurrent access for different objects or with no regard for objects is safe (i.e. no shared, global state), then all these functions here are, including those that add and remove callbacks. 

I wonder if `Yes` and `No` for values are enough. Maybe a third `Conditional` value would help us in being a bit more precise.

Reply:
Maybe @wjwwood can clarify how the `Thread-Safe` flag is used throughout the existing API.

Reply:
I don't really think we need a third state for Thread-Safe.  This one clearly isn't thread-safe; if you pass the same object into this method from two threads at the same time, that will leave the `clock` object in an undefined state.  So it looks to me like "Thread-Safe: No" is appropriate here.  The same goes for all of the APIs below.

Reply:
I thought the same, but it's also true that there're varying degrees of thread (un)safety. You can manage different clocks in different threads without issues. Functions that access global state don't even allow that.

Reply:
And I think that kind of differentiated information would be very valuable for the user.

Reply:
@wjwwood bump

Reply:
Sorry, I thought I replied to this, but maybe I was dreaming, I was talking about it with the Apex folks recently too.

The idea is that you can call it from multiple threads concurrently, but also generally that it isn't affected by other functions being called concurrently. If there's any threading considerations like "can be called concurrently but not on the same object" or "can be called concurrently but not concurrently with fini on the same object, etc...".

Basically, I consider the footnote to be more important than the Yes or No. If you say "Yes" without a note, that would mean to me that the function is most likely pure functional and has no global, or input/output, side effects. If it accesses global state or modifies an object passed by non-const pointer, then you should consider having a No or Yes with a note.

Reply:
Alright, so it could be either so long as there's a note. I'll add one.

Reply:
There, see ca78af86b84e3f7ab84ac08300aa341dc2a3e36a. It's a bit verbose but it is accurate.